specification/monitor/data-plane/Microsoft.Insights/preview/2023-05-01-preview/metricBatch.json

```
@@ -0,0 +1,535 @@
+      "x-ms-parameter-location": "method"
+    },
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "Client Api Version.",
+      "x-ms-parameter-location": "client"
+    }
+  },
+  "definitions": {
+    "ResourceIdList": {
+      "description": "The comma separated list of resource IDs to query metrics for.",
+      "type": "object",
+      "properties": {
+        "resourceids": {
+          "description": "The list of resource IDs to query metrics for.",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "format": "arm-id"
+          }
+        }
+      },
+      "example": {
+        "resourceids": [
+          "/subscriptions/123456728-abcd-1234-abcd-123456789abc/resourceGroups/TestGroup/providers/Microsoft.Storage/storageAccounts/TestStorage1",
+          "/subscriptions/123456728-abcd-1234-abcd-123456789abc/resourceGroups/TestGroup/providers/Microsoft.Storage/storageAccounts/TestStorage2"
+        ]
+      }
+    },
+    "AdditionalInfoErrorResponse": {
+      "description": "The response to a metrics query that results in a bad request, with optional additional information.",
+      "type": "object",
+      "properties": {
+        "error": {
+          "description": "Top level error object that contains all relevant information.",
+          "type": "object",
+          "properties": {
+            "additionalInfo": {
+              "description": "Additional information about the error",
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "type": {
+                    "description": "The type of the info property (e.g. string).",
+                    "type": "string"
+                  },
+                  "info": {
+                    "description": "Additional information related to the error.",
+                    "type": "string"
+                  }
+                }
+              }
+            },
+            "code": {
+              "description": "Error code",
+              "type": "string"
+            },
+            "message": {
+              "description": "Error message indicating why the operation failed.",
+              "type": "string"
+            }
+          }
+        }
+      }
```

The `error` property should be required (always present).
```suggestion
      },
      "required": ["error"]
```

---

Applied in the latest commit. Thank you!